pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -447,7 +447,21 @@ func (r *ReconcileBareMetalHost) actionDeleting(prov provisioner.Provisioner, in
 		info.log.Info("ready to be deleted")
 		return deleteComplete{}
 	}
-
+	// Try to power off host before delete except when it is externally provisioned.
+	if !info.host.Spec.ExternallyProvisioned {
+		provResult, err := prov.PowerOff()
+		switch {
+		case err != nil:
+			log.Error(err, fmt.Sprintf("Likely transient error while trying to Power Off Baremetal Host"))
+		case provResult.ErrorMessage != "":
+			log.Error(err, fmt.Sprintf("Error encountered while trying to Power Off Baremetal Host"))
+		case provResult.Dirty:
+			return actionContinue{provResult.RequeueAfter}
+		default:
+			info.host.Status.PoweredOn = false
```

Comment:
Should we return here to store the status update? Or are we relying on `prov.Delete()` returning dirty?

Reply:
Not sure. But, my thinking is that we should not return before attemping a prov.Delete() unless it is a general failure.

Reply:
Setting Dirty to true and returning here to save power status change.